src/instances/HeadcountInstance.ts

```diff
@@ -961,19 +1006,28 @@ export class HeadcountInstance {
                     return;
                 }
                 case HeadcountInstance.ABORT_HEADCOUNT_ID: {
-                    LOGGER.info(`${this._leaderName} aborted ${this._dungeon.dungeonName} headcount.`);
-                    this.abortHeadcount().then();
-                    return;
+                    if (i.user.id !== this._memberInit.id) {
+                        return this.showWrongLeaderbuttons(i);
+                    } else {
+                        LOGGER.info(`${(i.member as GuildMember).nickname} aborted ${this._dungeon.dungeonName} headcount.`);
+                        return this.abortHeadcount().then();
+                    }
                 }
                 case HeadcountInstance.DELETE_HEADCOUNT_ID: {
-                    LOGGER.info(`${this._leaderName} aborted ${this._dungeon.dungeonName} headcount after ending.`);
-                    this.abortHeadcount().then();
-                    return;
+                    if (i.user.id !== this._memberInit.id) {
+                        return this.showWrongLeaderbuttons(i);
+                    } else {
+                        LOGGER.info(`${(i.member as GuildMember).nickname} aborted ${this._dungeon.dungeonName} headcount after ending.`);
+                        return this.abortHeadcount().then();
+                    }
                 }
                 case HeadcountInstance.END_HEADCOUNT_ID: {
-                    LOGGER.info(`${this._leaderName} ended ${this._dungeon.dungeonName} headcount.`);
-                    this.endHeadcount().then();
-                    return;
+                    if (i.user.id !== this._memberInit.id) {
+                        return this.showWrongLeaderbuttons(i);
+                    } else {
+                        LOGGER.info(`${(i.member as GuildMember).nickname} ended ${this._dungeon.dungeonName} headcount.`);
+                        return this.abortHeadcount().then();
```

Comment:
Pressing the `End Headcount` button should ask the user if they want to end the headcount, not abort it.

When I press the `End Headcount` button on a headcount that doesn't belong to me, and then confirm that I want to end the headcount, it aborts the headcount.